libs/ml/examples/word2vec_distributed_learning/word2vec_client.hpp

```diff
@@ -244,6 +259,64 @@ void Word2VecClient<TensorType>::TestEmbeddings(std::string const &word0, std::s
   }
 }
 
+/**
+ * @return vector of gradient update values
+ */
+template <class TensorType>
+typename Word2VecClient<TensorType>::GradientType Word2VecClient<TensorType>::GetGradients()
+{
+  FETCH_LOCK(this->model_mutex_);
+  return GradientType(this->g_ptr_->GetGradients(), this->GetTimestamp(), this->id_,
+                      w2v_data_loader_ptr_->GetVocabHash());
+}
+
+/**
+ * @return pair of vocab strings and vocab hash
+ */
+template <class TensorType>
+std::pair<std::vector<std::string>, byte_array::ConstByteArray>
+Word2VecClient<TensorType>::GetVocab()
+{
+  auto vocab      = w2v_data_loader_ptr_->GetVocab();
+  auto vocab_hash = w2v_data_loader_ptr_->GetVocabHash();
+  return std::make_pair(vocab->GetReverseVocab(), vocab->GetVocabHash());
```

Comment:
according to the function names at least this GetVocab gets a ReverseVocab. That seems strange

Reply:
ReverseVocab is both the way to get a string from a word_id (i.e. the reverse of vocab which gets an id from a string), and also the most compact way of sending the information required for translation, so I'm not sure what to call it!

Reply:
it's just rather bizarre to see a function of the form
void GetX()
{
  return GetReverseX();
}
maybe add a comment for that last line to explain why its a good idea?